python/lsst/obs/base/yamlCamera.py

```diff
@@ -23,181 +23,318 @@
 
 import numpy as np
 import lsst.afw.cameraGeom as cameraGeom
+import lsst.geom as geom
 import lsst.afw.geom as afwGeom
 from lsst.afw.table import AmpInfoCatalog, AmpInfoTable
-from lsst.afw.cameraGeom.cameraFactory import makeDetector
+from lsst.afw.cameraGeom.cameraFactory import makeDetectorData
 
+__all__ = ["makeCamera"]
 
-class YamlCamera(cameraGeom.Camera):
-    """The Commissioning Camera (comCam)
+
+def makeCamera(cameraFile):
+    """An imaging camera (e.g. the LSST 3Gpix camera)
 
     Parameters
     ----------
-    cameraYamlFile : `str`
+    cameraFile : `str`
         Camera description YAML file.
+
+    Return a `lsst.afw.cameraGeom.Camera`
+    """
+
+    with open(cameraFile) as fd:
+        cameraParams = yaml.load(fd, Loader=yaml.Loader)
+
+    cameraName = cameraParams["name"]
```

Comment:
Where do we document the schema for these YAML files?

Reply:
We don't.  I think that that's out of scope for this set of changes, though.  File a ticket?